### DISCUSSION TITLE
Frigate in an unpriviliged LXC

### DISCUSSION THREAD

**Discussion by fbloemhof** [community-scripts/ProxmoxVE#930] [General]
By accident I installed the Frigate LXC as an unpriviliged container. I added an extra device to the config file to redirect my Coral TPU:

```
dev2: /dev/apex_0
```

And almost everything seems to work fine. The only significant error I currently see is 

```
Unable to poll intel GPU stats: Failed to initialize PMU! (Permission denied)
```

my guess this has something to do with the mapping of the GPU to the LXC and should be fixable.

So, are there any other reasons to run Frigate in a priviliged lxc?

**Comment by DanielCarmingham** [comment]
This issue has been bugging me so I dug into it...  

Changing the value of `kernel.perf_event_paranoid` to 0 on the host worked for me.

You can check the current value of your perf_event_paranoid flag by running this in the host shell:
```shell
sysctl -a | grep kernel.perf_event_paranoid
```

This will add it to your config and apply the change:
```shell
echo kernel.perf_event_paranoid=0 >> /etc/sysctl.d/local.conf
sysctl --system
```

This is mentioned in the Frigate docs [here](https://docs.frigate.video/configuration/hardware_acceleration/#perf_event_paranoid), but they recommend setting it to 2, which did not work for me.

### Related note:

On my host machine, the Intel iGPU will be at either `/dev/dri/card0` or `/dev/dri/card1` depending on if I have a monitor plugged in or not when booted.  This makes the LXCs that use it fail to start if it doesn't match the config.  

I removed the device mapping to `/dev/dri/cardX` entirely, but kept `/dev/dri/renderD128`, and it continued to use Intel QuickSync HW acceleration.  This worked for me both on my Docker+Frigate LXC and my Plex LXC.

I'd be curious to know if this works for other users and apps or if it causes problems.  If it works, we might not need the part of [this code](https://github.com/community-scripts/ProxmoxVE/blob/9523491c0ae3a71afb2d4ad872f6beb7176e7af5/misc/build.func#L785) that adds them.

HTH,
Daniel

**Reply by tehniemer** [reply]
> I removed the device mapping to /dev/dri/cardX entirely, but kept /dev/dri/renderD128, and it continued to use Intel QuickSync HW acceleration. This worked for me both on my Docker+Frigate LXC and my Plex LXC.

Thank you! This fixed my hw transcoding issues in Plex.

**Comment by fbloemhof** [comment]
Awesome, that indeed took away the error and it seems to work.

I don't recognize the card0/card1 behavior, it is always card1 at my machine.

**Reply by DanielCarmingham** [reply]
Interesting that you're not seeing the card0/card1 issue...  I'm guessing your hardware must be different.  Are you using a discrete GPU?

It happened on all 3 of the machines I've run Proxmox on with Intel iGPUs (2x N100-based minis and a i5-14500).

**Reply by fbloemhof** [reply]
My CPU is an i5-6500T and the GPU an "VGA compatible controller: Intel Corporation HD Graphics 530 (rev 06)".
